README.md

```diff
@@ -1,41 +1,138 @@
+
+### Supported Algorithms
+
+If an algorithm is provided by liboqs but is not listed below, it can still be used in the fork through [either one of two ways](https://github.com/open-quantum-safe/openssl/wiki/Using-liboqs-algorithms-that-are-not-in-the-forks).
+
+#### Key Exchange
+
+The following quantum-safe algorithms from liboqs are supported (assuming they have been enabled in liboqs):
+
+- `kemdefault` (see [here](https://github.com/open-quantum-safe/openssl/wiki/Using-liboqs-algorithms-that-are-not-in-the-forks#oqsdefault) for what this denotes)
+
+The following hybrid algorithms are supported only for L1 schemes; they combine an L1 quantum-safe algorithm listed above with ECDH that uses NIST's P256 curve:
+- `p256_<KEX>`, where ``<KEX>`` is any one of the L1 algorithms listed above.
+
+## Quickstart
+
+The steps below have been confirmed to work on Ubuntu 19.10 (gcc-8.3.0).
+
+### Building
+
+#### Linux
+
+#### Step 0: Get pre-requisites
+
+On **Ubuntu**, you need to install the following packages:
+
+	sudo apt install autoconf automake cmake gcc golang-go libtool libssl-dev make ninja unzip xsltproc
+
+Then, get source code of this fork (`<BORINGSSL_DIR>` is a directory of your choosing):
+
+	git clone --branch master https://github.com/open-quantum-safe/boringssl.git <BORINGSSL_DIR>
+
+#### Step 1: Build and install liboqs
+
+The following instructions will download and build liboqs, then install it into a subdirectory inside the OpenSSL folder.
+
+	git clone --branch master https://github.com/open-quantum-safe/liboqs.git
+	cd liboqs
+	autoreconf -i
+	./configure --prefix=<BORINGSSL_DIR>/oqs --without-openssl --enable-shared=no
+	make -j
+	make install
+
+#### Step 2: Build the fork
+
+Now we follow the standard instructions for building BoringSSL. Navigate to `<BORINGSSL_DIR>`, and:
+
+on **Ubuntu**, run:
+
+	mkdir build
+	cd build
+	cmake -GNinja ..
+	ninja
+
+The fork can also be built with shared libraries, to do so, run `cmake -DBUILD_SHARED_LIBRARIES=ON -GNinja ..`.
+
+To execute the white-box and black-box tests, run `ninja run_tests` from the `build` directory.
+
+### Running
+
+#### TLS demo
+
+BoringSSL contains a basic TLS server (`s_server`) and TLS client (`s_client`) which can be used to demonstrate and test TLS connections.
+
+To run a basic TLS server with all libOQS ciphersuites enabled, from the `build` directory, run:
+
+	tool/bssl s_server -accept 4433 -loop
+
+In another terminal window, you can run a TLS client requesting one of the supported ciphersuites (`<KEX>` = one of the quantum-safe or hybrid key exchange algorithms listed in the [Supported Algorithms](#supported-algorithms) section above):
+
+	tool/bssl s_client -curves oqs_<KEX> -connect localhost:4433
+
```

Comment:
Looks like the client and server can be specified with or without the `s_` prefix, and both options lead to the same functionality. 

That being said, I removed the prefix anyway since I think `client` and `server` read a bit better than `s_client` and `s_server`.